geopandas/geodataframe.py

```diff
@@ -494,9 +630,7 @@ def _dataframe_set_geometry(self, col, drop=False, inplace=False, crs=None):
     if inplace:
         raise ValueError("Can't do inplace setting when converting from"
                          " DataFrame to GeoDataFrame")
-    gf = GeoDataFrame(self)
-    # this will copy so that BlockManager gets copied
-    return gf.set_geometry(col, drop=drop, inplace=False, crs=crs)
```

Comment:
I have to think about this comment that is now removed .. :-) (whether it was important)